### PR TITLE
Glide.lock: Parse versions as Text

### PR DESCRIPTION
**Pull request by cnr**
We were previously parsing them as integers, which passed the tests from cliv1.

Updated the test `glide.lock` to add an alphanumeric hexadecimal version string for one of the dependencies

### CODE REVIEW

File: src/Strategy/Go/GlideLock.hs [hunk 1]
```diff
@@ -59,5 +60,9 @@ instance FromJSON GlideLockfile where
 instance FromJSON GlideDep where
   parseJSON = withObject "GlideDep" $ \obj ->
     GlideDep <$> obj .:  "name"
-               <*> obj .: "version"
+               -- version field can be text or an int (hexadecimal hash)
+               <*> (obj .: "version" <|> (intToText <$> obj .: "version"))
                <*> obj .:? "repo"
+
+intToText :: Int -> Text
+intToText = T.pack . show
```
Comment (scruffystuffs) [inline src/Strategy/Go/GlideLock.hs:68]: `Data.Text.Extra` has `showT :: Show a => a -> Text`

File: src/Strategy/Go/GlideLock.hs [hunk 2]
```diff
@@ -59,5 +60,9 @@ instance FromJSON GlideLockfile where
 instance FromJSON GlideDep where
   parseJSON = withObject "GlideDep" $ \obj ->
     GlideDep <$> obj .:  "name"
-               <*> obj .: "version"
+               -- version field can be text or an int (hexadecimal hash)
+               <*> (obj .: "version" <|> (intToText <$> obj .: "version"))
```
Comment (scruffystuffs) [inline src/Strategy/Go/GlideLock.hs:64]: Why are we not just parsing `Text`?  If we're validating input, we can just validate that text is digits or a hex hash, right?

Comment (cnr) [reply src/Strategy/Go/GlideLock.hs:64]: This is to handle the overwhelmingly-rare edge case where the hexadecimal number is a valid base-10 int (e.g., `1234` from the test file). In that case, it becomes a yaml `int`, so trying to parse it as `Text` would fail

Comment (scruffystuffs) [reply src/Strategy/Go/GlideLock.hs:64]: Another damn reason to hate yaml

Comment (cnr) [reply src/Strategy/Go/GlideLock.hs:64]: We're just not  p r a g m a t i c  enough. golang's yaml parser will happily coerce numbers to strings! (everything is a string if you squint hard enough)

**Review by scruffystuffs — APPROVED** [review]
Looks good.
